tests/e2e/browser/複数選択.spec.ts

```diff
@@ -0,0 +1,194 @@
+import { test, expect, Page } from "@playwright/test";
+import { toggleSetting, navigateToMain } from "../navigators";
+
+test.beforeEach(async ({ page }) => {
+  const BASE_URL = "http://localhost:5173/#/home";
+  await page.setViewportSize({ width: 800, height: 600 });
+  await page.goto(BASE_URL);
+
+  await navigateToMain(page);
+  await page.waitForTimeout(100);
+  await toggleSetting(page, "複数選択");
+
+  await addAudioCells(page, 3);
+});
+
+const ctrlLike = process.platform === "darwin" ? "Meta" : "Control";
+
+type SelectedStatus = {
+  active: number;
+  selected: number[];
+};
+/**
+ * アクティブなAudioCellと選択されているAudioCellを取得する。
+ * 戻り値のインデックスは1から始まる。（nth-childのインデックスと揃えるため）
+ */
+async function getSelectedStatus(page: Page): Promise<SelectedStatus> {
+  const selectedAudioKeys = await page.evaluate(() => {
+    const audioCells = [...document.querySelectorAll(".audio-cell")];
+    let active: number | undefined;
+    const selected: number[] = [];
+    for (let i = 0; i < audioCells.length; i++) {
+      const audioCell = audioCells[i];
+      if (audioCell.classList.contains("active")) {
+        active = i + 1;
+      }
+      if (audioCell.classList.contains("selected")) {
+        selected.push(i + 1);
+      }
+    }
+    if (active === undefined) {
+      throw new Error("No active audio cell");
+    }
+
+    return { active, selected };
+  });
+  return selectedAudioKeys;
+}
+
+async function addAudioCells(page: Page, count: number) {
+  for (let i = 0; i < count; i++) {
+    await page.getByRole("button", { name: "テキストを追加" }).click();
+    await page.waitForTimeout(100);
+  }
+}
+
+test("複数選択：マウス周り", async ({ page }) => {
+  let selectedStatus: SelectedStatus;
+
+  // 複数選択していない状態でactiveのAudioCellをクリックしても何も起こらない
+  await page.locator(".audio-cell:nth-child(1)").click();
+
+  await page.waitForTimeout(100);
+  selectedStatus = await getSelectedStatus(page);
+  expect(selectedStatus.active).toBe(1);
+  expect(selectedStatus.selected).toEqual([1]);
+
+  // Shift+クリックは前回選択していたAudioCellから今回クリックしたAudioCellまでを選択する
+  await page.locator(".audio-cell:nth-child(2)").click();
+  await page.keyboard.down("Shift");
+  await page.locator(".audio-cell:nth-child(4)").click();
+  await page.keyboard.up("Shift");
+
+  await page.waitForTimeout(100);
+  selectedStatus = await getSelectedStatus(page);
+  expect(selectedStatus.active).toBe(4);
+  expect(selectedStatus.selected).toEqual([2, 3, 4]);
+
+  // ただのクリックはactiveAudioKeyとselectedAudioKeysをクリックしたAudioCellだけにする
+  await page.locator(".audio-cell:nth-child(2)").click();
+
+  await page.waitForTimeout(100);
+  selectedStatus = await getSelectedStatus(page);
+  expect(selectedStatus.active).toBe(2);
+  expect(selectedStatus.selected).toEqual([2]);
+
+  if (process.platform === "darwin") {
+    // なぜかCmd(Meta)+クリックが動かないのでスキップする
+    // FIXME: 動くようにする
+    return;
+  }
```

Comment:
ちょっと細かいですが、これに気づかずmac環境でここ以降のテスト書き変えてなぜかCIでだけ動かない、みたいな悲劇がありそうに感じました。
Github Actions上でだけ動かないようにしておくのが良いかも。

```suggestion
  if (process.platform === "darwin" && !!process.env.CI) {
    // なぜかCmd(Meta)+クリックが動かないのでスキップする
    // FIXME: 動くようにする
    return;
  }
```